assets/js/googlesitekit/widgets/components/WidgetRecoverableModules.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * WidgetRecoverableModules component.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import useWidgetStateEffect from '../hooks/useWidgetStateEffect';
+import RecoverableModules from '../../../components/RecoverableModules';
+
+// The supported props must match `RecoverableModules` (except `widgetSlug`).
+export default function WidgetRecoverableModules( {
+	widgetSlug,
+	moduleSlugs,
+	...props
+} ) {
+	const metadata = useMemo(
+		() => ( {
+			// Here we serialize to `moduleSlug` for compatibility with the logic in
+			// `combineWidgets()`. In future we may wish to take a less "hacky" approach.
+			// See https://github.com/google/site-kit-wp/issues/5376#issuecomment-1165771399.
+			moduleSlug: JSON.stringify( moduleSlugs.sort() ),
```

Comment:
We shouldn't call `sort` on the `moduleSlugs` prop directly because it will mutate the input.

Also, let's use `join` instead of stringify because the result for a module that only relies on a single module will be only the one module's slug rather than a JSON string of the array.

```suggestion
			moduleSlug: [ ...moduleSlugs ].sort().join(),
```

Reply:
Thanks, good spot on the `sort` mutation!

I've fixed that and used `join` rather than `JSON.stringify`. I joined using a comma to make the value a bit more coherent and avoid an obscure edge case...

Reply:
Great thanks! `join()` uses a comma by default but good to be explicit I suppose. You can also do `[1,2,3].toString()` which will be comma-separated as well 😉

Reply:
Well, thanks for another reminder of the basics. Mutable `sort` was an oversight on my part, but I had completely forgotten `join` defaults to a comma! Regardless, I do prefer the explicit version (probably evident from the fact I had forgotten about the default), for the sake of, well, being explicit. :)